### ISSUE TITLE
app to sync asus aura and razer chroma (exists already)

### ISSUE THREAD

**Issue by fudoido** [poveden/EliteChroma#90]
today i did a bit of research and found something that can make this possible.
https://github.com/ChromaControl/ChromaControl/tree/msix-beta

as being new, it is limited and might have alot of bugs, but i still decided to try it.
asus b450 plus motherboard, Crucial Ballistix Tactical Tracer memory, razer cynosa v2 keyboard.
motherboard only has normal rgb strips, not addressable, memory is addressable along the length of the dim.
motherboard and ram show all solid light blue color when in space, light greenish when docked, but when entering superjump or hyperjump, the strips flash with some randomness, or i just was not able to get the right sequence of them, and ram flashes the leds randomly across the dim, but not all at the same time, as if it was like the jump animation on the cynosa keyboard.

seeing that this is a new software, made by a common member of the github community (he has made a previous one for aura, however this new one has a "library" for aura and another one for corsair separated), i wonder how it  can be used to work with elite chroma.......
just to add, in the synapse software, the "chromacontrol" software shows up as a single device even when there might be more in the pc. in my case, 2 rgb strips, the amd wraith rgb cooler (not the addressable one) and 2 dims of ram, but synapse detects all this as a single device using the chroma control software. nontheless, its something to keep an eye and expand functionality of elite chroma hopefully, maybe make it work also when scooping, as an example (i feel like it reads some of the keyboard keys only and shows whatever color is on them, so when animations come and keys get off, strips go off also and part of the ram).

When possible, i will make a video for youtube and try to add the link on a reply, so that the animations can be seen better than explained, if there is interest in checking this software before using it.
Regards.

**Comment by poveden** [comment]
Hi! Thanks for sharing this. Right now I'm focused on consuming the Razer Chroma SDK, but I do not discard adding support to other technologies in the future. It will depend on my available time, and a willing "guinea pig" 😛 for testing (as I don't see myself purchasing Asus hardware anytime soon).

Let me briefly explain how EliteChroma handles Chroma Link devices (e.g. Philips Hue, Nanoleaf, or anything pluggable into Razer Synapse):

- [Chroma Link exposes 5 addressable "LEDs" (`CL1` through `CL5`)](https://developer.razer.com/works-with-chroma/chroma-link-guide/).
- EliteChroma "imagines" `CL2`-`CL5` as being "around" the ship, like this:
  ```
   CL3 /\ CL4
  CL2 /__\ CL5
  ```
- `CL1` acts as an "ambient" color, and will reflect the average of `CL2`-`CL5`.
- When a light particle "passes" next to the ship, `CL3`->`CL2` or `CL4`->`CL5` will be lighted up in sequence (depending if passing to the ship's left or right, respectively), trying to simulate the particle's movement through space.

Unless the user configures correctly the lights on Razer Synapse, the intended effect surely will appear as random light flickering. I'm afraid that I don't have a setup to properly test this (just a pair of Philips light bulbs and a light strip), so any suggestions or contributions are more than welcome!

**Comment by fudoido** [comment]
i would say that the ram has all those "cl" adresses, but strip is only 1 adress as it only lights in a couple occasions during the supercruise/hyperjump.
it would be great if the fuel scooping animation worked also with it, but not at the time being. not sure why only the starfield effect runs in the ram and strip, but at least there is some animations thanks to the work of the other member to connect synapse with aura equipment. :)

**Comment by poveden** [comment]
If the strip lights entirely with only 1 color, then indeed it's only being addressed by the `CL1` LED. Can you confirm?

I'll take note about making ChromaLink devices pulse during fuel scooping.

**Comment by fudoido** [comment]
if cl1 is meant to be an ambient colour, it should be on all the time?
if so, that is not the case, it turn on and off while doing the animation.
i do notice that while in supercruise, all leds are blue with just some soft variations in intensity, off during normal speed, and animated while doing the charge of SC/HJ and travel of the HJ ingame.

**Comment by fudoido** [comment]
2 videos just recorded, but you might have to download them because the fastest way i had to upload directly from my phone was to my onedrive.
1st video is how the other app is recognized in razer synapse. seems to identify it as one device with 4 channels, as it shows 4 blinking dots in the device.
[https://1drv.ms/v/s!AkXBtY_Q3jJGrhmSj_t3hXRbfowW?e=z9fwfA](url)

2nd video is how it is working along with elitechroma
[https://1drv.ms/v/s!AkXBtY_Q3jJGrhhN-sJb93b919eE?e=8p6G0h](url)

**Comment by poveden** [comment]
@fudoido Could you please re-share? I cannot open those files.

**Comment by fudoido** [comment]
aura sync app with synapse demo - https://1drv.ms/v/s!AkXBtY_Q3jJGrhmSj_t3hXRbfowW?e=SueXCY

Aura sync app working with elitechroma - https://1drv.ms/v/s!AkXBtY_Q3jJGrhhN-sJb93b919eE?e=YyYYej

Hope you get the files this time, and sorry for the late reply, been a bit distracted lately due work tiredness.

**Comment by poveden** [comment]
I finally got the time to see your videos. Your setup lights as expected. Your LEDs are greenish because that's the color you have set up for the ship's HUD (instead of the regular orange color).

During fuel scoop I decided _not_ to light everything, but just the keyboard, so it's behaving as expected as well.

I will, however, make the following change: Instead of treating the lights' disposition as
```
↓  CL3 /\ CL4  ↓
↓ CL2 /__\ CL5 ↓
```
which makes the "stars" in hyperspace "flow" like `CL3→CL2` and `CL4→ CL5`, I will arrange them as
```
↓    CL2 /\ CL2    ↓
↓   CL3 /  \ CL3   ↓
↓  CL4 /    \ CL4  ↓
↓ CL5 /______\ CL5 ↓
```
so the flow goes like `CL2→CL3→CL4→CL5`. It should be more fluid and easy on the eyes. I'll let you know when I have it.

**Comment by fudoido** [comment]
only today i read your reply, been on holidays and enjoying a bit away from computer ^^
i have been using the synapse ambilight feature more recently, i can select parts of the screen to capture which color will light up and it has provided a bigger immersion while playing elite, if that region goes black, my leds go off, if i face a star, it all goes very bright, and tries to match up the colors as much as possible.
i selected the area where the supercruise/jump bar charges and it gets quite immersive also to see the case leds light up and change color as the bar fills up.
besides the razer keyboard and the led strips and fans in my case, connected to a msi motherboard and compatible with synapse, i also bought a led bulb from yeelight, cheap and also compatible with synapse, so now the light immersion fills up my entire room, not just the case and keyboard :P 
wish i could use just the bulb with synapse "ambilight" feature and everything else with the elitechroma, but as you mentioned before, addon softwares for synapse take over control of everything, not just accessory A or B, as seen with other games as well (have a couple games that have synapse compability built in them from developers, and the games also take over everything).

**Comment by poveden** [comment]
> ... so the flow goes like CL2→CL3→CL4→CL5. It should be more fluid and easy on the eyes. _I'll let you know when I have it._

@fudoido the change of hyperspace flows on ChromaLink devices is now available since release v1.13.0. Please let me know if it looks better now.

Since this discussion is more about Aura than anything else, I'm closing it now. Feel free to answer here or open a new issue if needed. Thanks for your feedback.
